include/swift/Sema/IDETypeChecking.h

```diff
@@ -60,11 +60,20 @@ namespace swift {
   /// Typecheck binding initializer at \p bindingIndex.
   void typeCheckPatternBinding(PatternBindingDecl *PBD, unsigned bindingIndex);
 
+  /// Attempt to merge two types for the purposes of lookup. In general this
+  /// means preferring a subtype over a supertype, but can also e.g prefer an
+  /// optional over a non-optional. If the two types are incompatible, null is
+  /// returned.
+  Type tryMergeBaseTypeForLookup(Type ty1, Type ty2, DeclContext *dc);
```

Comment:
Is this is usable other than code completion? If not, maybe `tryMergeBaseTypeForCompletionLookup`